posthog/hogql/database/database.py

```diff
@@ -167,6 +168,11 @@ def create_hogql_database(team_id: int, modifiers: Optional[HogQLQueryModifiers]
         database.events.fields["poe"].fields["id"] = database.events.fields["person_id"]
         database.events.fields["person"] = FieldTraverser(chain=["poe"])
 
+    database.persons.fields["$initial_referring_domain_type"] = create_initial_domain_type(
+        "$initial_referring_domain_type"
+    )
+    database.persons.fields["$initial_channel_type"] = create_initial_channel_type("$initial_channel_type")
```

Comment:
I also now wonder if we want the `$` here. It looks like a property, but isn't. Perhaps we should use some other way of namespacing these? `__initial_...` or such? I can imagine there will be a bit of confusion if it looks like a property, but you can't find it under properties, for e.g. filters and such.

Alternatively, maybe we do want to bring forward the idea of actual calculated properties, property schemas (type verification), and so forth? This way we could expose `original_properties`, and have a new magical `properties` column that "inherits" from it?

I realise this is a textbook example of scope creep though, and definitely not blocking this

Reply:
Changed - I used $virt to prefix the virtual columns